opencsp/common/lib/tool/tk_tools.py

```diff
@@ -0,0 +1,123 @@
+        # Leaves only the label and removes the app window
+        self.tw.wm_overrideredirect(True)
+        self.tw.wm_geometry(f"+{x:d}+{y:d}")
+        label = tk.Label(
+            self.tw,
+            text=self.text,
+            justify='left',
+            background="#ffffff",
+            relief='solid',
+            borderwidth=1,
+            wraplength=self.wraplength,
+        )
+        label.pack(ipadx=1)
+
+    def hidetip(self):
+        tw = self.tw
+        self.tw = None
+        if tw:
+            tw.destroy()
+
+
+def _window(*vargs, TopLevel=False, **kwargs):
+    """Initializes and returns a new tkinter.Tk (or tkinter.TopLevel) instance.
+
+    If creating the window fails, tries again (up to two more times).
+
+    Sometimes initializing a tk window fails. But if you try to
+    initialize the window again, it seems to always succeed.
+    When it fails, it is often with an error about not being able to find a
+    file. Something like::
+
+        _tkinter.TclError: Can't find a usable init.tcl in the following directories
+    """
+    window_class = tk.Tk
+    if TopLevel:
+        window_class = tk.Toplevel
+
+    try:
+        # try to create a window
+        return window_class(*vargs, **kwargs)
+    except Exception:
+        try:
+            lt.warn("Failed to create a tkinter.Tk() window. Trying again (2nd attempt).")
+            # first attempt failed, try again
+            return window_class(*vargs, **kwargs)
+        except Exception:
+            # second attempt failed, give the system a second to stabalize and
+            # try a third time
+            lt.warn("Failed to create a tkinter.Tk() window. Trying again (3rd attempt).")
+            import time
+
+            time.sleep(1)
+            return window_class(*vargs, **kwargs)
+
+
+def _window(*vargs, **kwargs):
+    """Initializes and returns a new tkinter.Tk() instance.
+
+    If creating the window fails, tries again (up to two more times).
+
+    Sometimes initializing a tk window fails. But if you try to
+    initialize the window again, it seems to always succeed.
+    When it fails, it is often with an error about not being able to find a
+    file. Something like::
+
+        _tkinter.TclError: Can't find a usable init.tcl in the following directories
+    """
+    try:
+        # try to create a window
+        return tk.Tk(*vargs, **kwargs)
+    except Exception:
+        try:
+            lt.warn("Failed to create a tkinter.Tk() window. Trying again (2nd attempt).")
+            # first attempt failed, try again
+            return tk.Tk(*vargs, **kwargs)
+        except Exception:
+            # second attempt failed, give the system a second to stabalize and
+            # try a third time
+            lt.warn("Failed to create a tkinter.Tk() window. Trying again (3rd attempt).")
+            import time
+
+            time.sleep(1)
+            return tk.Tk(*vargs, **kwargs)
```

Comment:
How does it work (mostly for my benefit) with having two functions with the same name?
Also, why does it a private function (with an underscore in front)? Should it not just be `def window(...)`?

Reply:
Yeah, this was a copy-paste mistake. I also found this error while responding to these comments.

You're also right that it shouldn't be private. Fixed!